libbeat/common/kubernetes/watcher.go

```diff
@@ -204,6 +204,11 @@ func (w *watcher) process(ctx context.Context) bool {
 			return nil
 		}
 		if !exists {
+			if entry.state == delete {
+				w.logger.Errorf("Object was not found in the store, deleting anyway!")
```

Comment:
If this happens too frequently this can flood logs and maybe the user cannot do anything about that. Wdyt about logging this at the debug level? Maybe logging also the object or some info about it.